imblearn/utils/_validation.py

```diff
@@ -27,6 +27,42 @@
 TARGET_KIND = ("binary", "multiclass", "multilabel-indicator")
 
 
+class _OutputReconstructor:
+    """A class for converting input types to numpy and back."""
+
+    def __init__(self, X, y):
```

Comment:
Do you think that we could pass `__init__(self, *arrays)`. Like this we could imagine to `sample_weight` maybe?

Reply:
Do we know already what to do with `sample_weight`?

Reply:
We just know that we should sample them at some point :)